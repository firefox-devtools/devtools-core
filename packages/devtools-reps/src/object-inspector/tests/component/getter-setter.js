/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

const { mount } = require("enzyme");
const React = require("react");
const { createFactory } = React;
const ObjectInspector = createFactory(require("../../index"));
const { MODE } = require("../../../reps/constants");

const accessorStubs = require("../../../reps/stubs/accessor");

describe("ObjectInspector - getters & setters", () => {
  it("renders getters as expected", () => {
    const stub = accessorStubs.get("getter");
    const oi = mount(ObjectInspector({
      autoExpandDepth: 1,
      roots: [{
        path: "root",
        name: "x",
        contents: stub
      }],
      getObjectProperties: () => {},
      loadObjectProperties: () => {},
      mode: MODE.LONG,
    }));

    const nodes = oi.find(".node");
    // There should be the root and a leaf.
    expect(nodes.length).toBe(2);

    // The root should be expandable.
    const rootLeaf = nodes.first();
    expect(rootLeaf.find(".arrow").exists()).toBeTruthy();
    expect(rootLeaf.text()).toBe("x : Getter");

    const getLeaf = nodes.last();
    expect(getLeaf.find(".arrow").exists()).toBeTruthy();
    expect(getLeaf.text()).toBe("<get> : function get x()");
  });

  it("renders setters as expected", () => {
    const stub = accessorStubs.get("setter");
    const oi = mount(ObjectInspector({
      autoExpandDepth: 1,
      roots: [{
        path: "root",
        name: "x",
        contents: stub
      }],
      getObjectProperties: () => {},
      loadObjectProperties: () => {},
      mode: MODE.LONG,
    }));

    const nodes = oi.find(".node");
    // There should be the root and a leaf.
    expect(nodes.length).toBe(2);

    // The root should be expandable.
    const rootLeaf = nodes.first();
    expect(rootLeaf.find(".arrow").exists()).toBeTruthy();
    expect(rootLeaf.text()).toBe("x : Setter");

    const getLeaf = nodes.last();
    expect(getLeaf.find(".arrow").exists()).toBeTruthy();
    expect(getLeaf.text()).toBe("<set> : function set x()");
  });

  it("renders getters and setters as expected", () => {
    const stub = accessorStubs.get("getter setter");
    const oi = mount(ObjectInspector({
      autoExpandDepth: 1,
      roots: [{
        path: "root",
        name: "x",
        contents: stub
      }],
      getObjectProperties: () => {},
      loadObjectProperties: () => {},
      mode: MODE.LONG,
    }));

    const nodes = oi.find(".node");
    // There should be the root and 2 leaves.
    expect(nodes.length).toBe(3);

    // The root should be expandable.
    const rootLeaf = nodes.first();
    expect(rootLeaf.find(".arrow").exists()).toBeTruthy();
    expect(rootLeaf.text()).toBe("x : Getter & Setter");

    const getLeaf = nodes.at(1);
    expect(getLeaf.find(".arrow").exists()).toBeTruthy();
    expect(getLeaf.text()).toBe("<get> : function get x()");

    const setLeaf = nodes.at(2);
    expect(setLeaf.find(".arrow").exists()).toBeTruthy();
    expect(setLeaf.text()).toBe("<set> : function set x()");
  });
});
