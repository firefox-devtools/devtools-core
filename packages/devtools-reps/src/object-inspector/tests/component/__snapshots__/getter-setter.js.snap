// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectInspector - getters & setters renders getters and setters as expected 1`] = `
"▼ x: Getter & Setter
|  ▶︎ <get>: function get x()
|  ▶︎ <set>: function set x()"
`;

exports[`ObjectInspector - getters & setters renders getters as expected 1`] = `
"▼ x: Getter
|  ▶︎ <get>: function get x()"
`;

exports[`ObjectInspector - getters & setters renders setters as expected 1`] = `
"▼ x: Setter
|  ▶︎ <set>: function set x()"
`;
